### PR TITLE
Remove calories from trip details

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
Removes calorie counts and repurposes all methods with "Time Active" which combines time spent biking and walking to reflect the total number of minutes spent active in the trip duration.

image

### CODE REVIEW

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I recommend not changing anything in core-utils.
Please fill in intl content for the desired languages.

I would prefer the activity time to be a new feature instead of a breaking change, but if people feel strongly about removing calories, please add a breaking change commit.

File: packages/core-utils/src/itinerary.ts [hunk 1]
```diff
@@ -412,22 +412,20 @@ export function getTNCLocation(leg: Leg, type: string): string {
 
 export function calculatePhysicalActivity(
   itinerary: ItineraryOnlyLegsRequired
-): {
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/core-utils/src/itinerary.ts:415]: Caution: If this change is left in this PR, the trip-details package will still reference the currently used core-utils (8.1.1), even though core-utils version 9.0.0 will be released. This should not break the trip-details package.

File: packages/trip-details/src/index.tsx [hunk 2]
```diff
@@ -3,20 +3,20 @@ import coreUtils from "@opentripplanner/core-utils";
...
 import TripDetail from "./trip-detail";
 import FareLegTable from "./fare-table";
 import { boldText, renderFare } from "./utils";
 
 import {
-  CaloriesDetailsProps,
   TransitFareProps,
-  TripDetailsProps
+  TripDetailsProps,
+  TimeActiveDetailsProps
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.tsx:19]: Sort imports.

File: packages/trip-details/src/TripDetails.story.tsx [hunk 3]
```diff
@@ -377,7 +372,7 @@ export const TncTransitItinerary = makeStory(
 
 export const TncTransitItineraryWithCustomMessages = makeStory(
   {
-    CaloriesDetails: CustomCaloriesDetails,
+    TimeActiveDetails: CustomTimeActiveDetails,
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/TripDetails.story.tsx:375]: Sort attributes.

File: packages/trip-details/src/TripDetails.story.tsx [hunk 4]
```diff
@@ -249,7 +244,7 @@ function createTripDetailsTemplate(
       : {};
     return (
       <Component
-        CaloriesDetails={CaloriesDetails}
+        TimeActiveDetails={TimeActiveDetails}
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/TripDetails.story.tsx:247]: Sort props.

File: packages/trip-details/src/TripDetails.story.tsx [hunk 5]
```diff
@@ -232,7 +227,7 @@ function createTripDetailsTemplate(
 ): ComponentStory<typeof TripDetails> {
   const TripDetailsTemplate = (
     {
-      CaloriesDetails,
+      TimeActiveDetails,
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/TripDetails.story.tsx:230]: Sort attributes.

File: packages/trip-details/src/TripDetails.story.tsx [hunk 6]
```diff
@@ -13,7 +13,7 @@ import styled from "styled-components";
 import TripDetails, { FareLegTable } from ".";
 import * as TripDetailsClasses from "./styled";
 import {
-  CaloriesDetailsProps,
+  TimeActiveDetailsProps,
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/TripDetails.story.tsx:16]: Sort imports.

File: packages/trip-details/src/index.tsx [hunk 7]
```diff
@@ -111,13 +93,13 @@ const TransitFare = ({
 };
 
 /**
- * Renders trip details such as departure instructions, fare amount, and calories spent.
+ * Renders trip details such as departure instructions and fare amount.
  */
 export function TripDetails({
-  CaloriesDetails = DefaultCaloriesDetails,
+  TimeActiveDetails = DefaultTimeActiveDetails,
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.tsx:99]: Sort props

File: packages/trip-details/src/types.ts [hunk 8]
```diff
@@ -63,9 +63,9 @@ export interface TransitFareProps {
 
 export interface TripDetailsProps {
   /**
-   * Slot for a custom component to render the expandable section for calories.
+   * Slot for a custom component to render the expandable section for time active.
    */
-  CaloriesDetails?: React.ElementType<CaloriesDetailsProps>;
+  TimeActiveDetails?: React.ElementType<TimeActiveDetailsProps>;
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/types.ts:68]: Sort props.

File: packages/trip-details/i18n/en-US.yml [hunk 9]
```diff
@@ -23,7 +16,12 @@ otpUi:
       special: Special
       youth: Youth
     hideDetail: Hide details
+    minutesActive: "Time Spent Active: {minutes, plural, one {# minute} other {# minutes}}"
     showDetail: Show details
+    timeActiveDescription: >
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/i18n/en-US.yml:21]: `timeActiveDescription` needs to be provided in the other languages.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please see the cleanup and missing translation comments. I would suggest making the core-utils changes in a separate PR.

File: packages/core-utils/src/itinerary.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply packages/core-utils/src/itinerary.ts]: Thanks for explaining this! I've re-added this function to core-utils.

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/core-utils/src/itinerary.ts]: Hmm... Commit https://github.com/opentripplanner/otp-ui/pull/570/commits/bdf0531725443610764472af246826d2d192a917 was not reverted (using a commitizen `revert` commit), so in order to not gamble on the outcome of this PR, I would recommend you create a new PR with just the changes to the `trip-details` package.

**Comment by amy-corson-ibigroup** [comment]
Closed in favor of #580

**Pull request closed without merging**